### ISSUE TITLE
Select

### ISSUE THREAD

**Issue by inumanag** [ocaml/dune#1481]
Is it possible to pass a custom flags for specific modes? For example, the dune file below works for `dune build main.exe` but fails for `dune build main.exe.o` because `ocamlopt` does not expect `-lfoo`. Is there a way to set `-lfoo` only for exe mode and not for object mode?

Problematic example:
```
(executable
   (name main)
   (libraries base core ctypes ctypes.foreign ctypes.stubs)
   (link_flags (-cclib -lfoo))
   (modes object exe))
```

Thanks!

**Comment by rgrinberg** [comment]
You can do what you did and simply copy the module. There's a `(copy# ..)` action for this purpose.

**Comment by nojb** [comment]
A cleaner approach depending on your situation may be to define a `library` with the modules that depend on `-lfoo`, passing `-cclib -lfoo` in the `library_flags` field. Then there should be no need to pass this flag when creating the executable.

**Comment by emillon** [comment]
@voodoos does #5649 fix this?

**Comment by voodoos** [comment]
> @voodoos does #5649 fix this?

No, #5649 allows mode-dependent flags when building foreign stubs only, not ocaml flags.
